samples/JwtBearerSample/Startup.cs

```diff
@@ -86,13 +67,16 @@ public void Configure(IApplicationBuilder app)
             app.Use(async (context, next) =>
             {
                 // Use this if there are multiple authentication schemes
-                // var user = await context.Authentication.AuthenticateAsync(JwtBearerDefaults.AuthenticationScheme);
-
-                var user = context.User; // We can do this because of there's only a single authentication scheme
-                if (user?.Identity?.IsAuthenticated ?? false)
+                var authResult = await context.AuthenticateAsync(JwtBearerDefaults.AuthenticationScheme);
+                if (authResult.Succeeded && authResult.Principal.Identity.IsAuthenticated)
                 {
                     await next();
                 }
+                else if (authResult.Failure != null)
+                {
+                    // Rethrow, let the exception page handle it.
+                    ExceptionDispatchInfo.Capture(authResult.Failure).Throw();
```

Comment:
Do you really want to encourage developers to throw an exception for non-exceptional things like expired tokens? :sweat_smile:

Reply:
For debugging 😁

Reply:
Without the usual "don't do that in a real world app", I'm convinced folks will blindly copy this snippet without realizing it's not really a good idea :smile: